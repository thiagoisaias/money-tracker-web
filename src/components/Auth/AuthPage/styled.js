import styled from "styled-components";
import { devices } from "utils/devices";

import image from "assets/images/welcome.jpg";

export const Wrapper = styled.div`
  display: flex;
  align-items: center;
  height: 100vh;
`;

export const FormWrapper = styled.div`
  @media ${devices.mediumDown} {
    margin: 0 auto;
    width: 100%;
    max-width: 400px;
    padding: 0 32px;
  }

  @media ${devices.largeUp} {
    width: 40%;
    max-width: 350px;
    margin: 0 auto;
    padding: 0 32px;
  }
`;

export const PictureWrapper = styled.div`
  color: #eee;
  position: relative;

  @media ${devices.mediumDown} {
    display: none;
  }

  @media ${devices.largeUp} {
    background: linear-gradient(rgba(0, 0, 0, 0.15), rgba(0, 0, 0, 0.15)),
      url(${image}) center no-repeat;
    background-size: cover;
    width: 60%;
    height: 100%;
    display: flex;
    justify-content: center;
    flex-direction: column;
  }
`;

export const PictureTitle = styled.h2`
  font-size: 2.5em;
  letter-spacing: 1px;
  margin-bottom: 16px;
  font-weight: 600;
  margin-left: 55px;
`;

export const PictureSubtitle = styled.p`
  font-size: 16px;
  color: #ddd;
  opacity: 0.9;
  margin-left: 57px;
`;

export const PhotoCredit = styled.span`
  position: absolute;
  color: #ddd;
  font-size: 12px;
  bottom: 16px;
  right: 8px;
  cursor: default;
`;
