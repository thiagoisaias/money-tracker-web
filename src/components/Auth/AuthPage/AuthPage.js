import React from "react";
import { Route, Switch } from "react-router-dom";
import styled from "styled-components";
import { devices } from "../../../utils/devices";

import image from "../../../assets/images/welcome.jpg";
import LoginFormContainer from "../../../containers/Auth/Login/LoginFormContainer";
import SignupFormContainer from "../../../containers/Auth/Signup/SignupFormContainer";

const Container = styled.div`
  display: flex;
  align-items: center;
  height: 100vh;
`;

const FormContainer = styled.div`
  @media ${devices.mediumDown} {
    margin: 0 auto;
    width: 100%;
    max-width: 400px;
    padding: 0 32px;
  }

  @media ${devices.largeUp} {
    width: 40%;
    max-width: 350px;
    margin: 0 auto;
    padding: 0 32px;
  }
`;

const PictureContainer = styled.div`
  color: #eee;
  position: relative;

  @media ${devices.mediumDown} {
    display: none;
  }

  @media ${devices.largeUp} {
    background: linear-gradient(rgba(0, 0, 0, 0.15), rgba(0, 0, 0, 0.15)),
      url(${image}) center no-repeat;
    background-size: cover;
    width: 60%;
    height: 100%;
    display: flex;
    justify-content: center;
    flex-direction: column;
  }
`;

const PictureTitle = styled.h2`
  font-size: 2.5em;
  letter-spacing: 1px;
  margin-bottom: 16px;
  font-weight: 600;
  margin-left: 55px;
`;

const PictureSubtitle = styled.p`
  font-size: 16px;
  color: #ddd;
  opacity: 0.9;
  margin-left: 57px;
`;

const PhotoCredit = styled.span`
  position: absolute;
  color: #ddd;
  font-size: 12px;
  bottom: 16px;
  right: 8px;
  cursor: default;
`;

export const AuthPage = props => {
  return (
    <Container>
      <FormContainer>
        <Switch>
          <Route exact path="/auth/" component={SignupFormContainer} />
          <Route exact path="/auth/signup" component={SignupFormContainer} />
          <Route exact path="/auth/login" component={LoginFormContainer} />
        </Switch>
      </FormContainer>
      <PictureContainer>
        <PictureTitle>{"Welcome to Wallet"}</PictureTitle>
        <PictureSubtitle>
          {"Lorem ipsum dolor sit amet, consectetur adipiscing elit"}
        </PictureSubtitle>
        <PhotoCredit>{"Photo by Romello Williams on Unsplash"}</PhotoCredit>
      </PictureContainer>
    </Container>
  );
};

export default AuthPage;
