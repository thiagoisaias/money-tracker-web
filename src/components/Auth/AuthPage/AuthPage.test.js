import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import AuthPage from "./AuthPage";

describe("AuthPage", () => {
  const component = shallow(<AuthPage />);

  it("renders properly", () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
