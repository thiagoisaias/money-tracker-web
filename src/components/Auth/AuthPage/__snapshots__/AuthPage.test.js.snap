// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthPage renders properly 1`] = `
<styled.div>
  <styled.div>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/auth/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/auth/signup"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/auth/login"
      />
    </Switch>
  </styled.div>
  <styled.div>
    <styled.h2>
      Welcome to Wallet
    </styled.h2>
    <styled.p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit
    </styled.p>
    <styled.span>
      Photo by Romello Williams on Unsplash
    </styled.span>
  </styled.div>
</styled.div>
`;
